backend-rust/src/graphql_api/baker.rs

```diff
@@ -350,6 +352,79 @@ impl Baker {
+        // is defined to be the minimum of the capital bound cap and the
+        // leverage bound cap:
+
+        // leverage bound cap for pool p: Lₚ = λ * Cₚ - Cₚ = (λ - 1) * Cₚ
+        // capital bound cap for pool p: Bₚ = floor( (κ * (T - Dₚ) - Cₚ) / (1 - K) )
+
+        // Where
+        // κ is the capital bound
+        // λ is the leverage bound
+        // T is the total staked capital on the whole chain (including passive
+        // delegation)
+        // Dₚ is the delegated capital of pool p
+        // Cₚ is the equity capital (staked by the pool owner excluding delegated stake
+        // to the pool) of pool p
+
+        // The `leverage bound cap` ensures that each baker has skin
+        // in the game  with respect to its delegators by providing some of the CCD
+        // staked from its own funds. The `capital bound cap` helps maintain
+        // network decentralization by preventing a single baker from gaining
+        // excessive power in the consensus protocol.
+
+        let current_chain_parameters = sqlx::query_as!(
+            DelegatedStakeBounds,
+            "
+            SELECT 
+                leverage_bound_numerator,
+                leverage_bound_denominator,
+                capital_bound 
+            FROM current_chain_parameters 
+            WHERE id = true
+            "
+        )
+        .fetch_one(pool)
+        .await?;
+
+        let leverage_bound: f64 = current_chain_parameters.leverage_bound_numerator as f64
+            / current_chain_parameters.leverage_bound_denominator as f64;
+        let leverage_bound_cap_for_pool: Amount = ((leverage_bound - 1f64) * self.staked as f64)
+            .try_into()
+            .map_err(|e: anyhow::Error| ApiError::InternalError(e.to_string()))?;
+
+        // The value is stored as a fraction with precision of `1/100_000` in the
+        // database.
+        let capital_bound: f64 = current_chain_parameters.capital_bound as f64 / 100000f64;
+
+        let capital_bound_cap_for_pool: Amount = if capital_bound == 1f64 {
+            // To avoid dividing by 0 in the `capital bound cap` formula,
+            // we only apply the `leverage_bound_cap_for_pool` in that case.
+            leverage_bound_cap_for_pool
+        } else {
+            let capital_bound_cap_for_pool_numerator: f64 = capital_bound
+                * ((total_stake - delegated_stake_of_pool) as f64)
+                - self.staked as f64;
+
+            // Denominator is not zero since we checked that `capital_bound != 1`.
+            let capital_bound_cap_for_pool_denominator: f64 = 1f64 - capital_bound;
+
+            let capital_bound_cap_for_pool: f64 =
+                capital_bound_cap_for_pool_numerator / capital_bound_cap_for_pool_denominator;
+
+            capital_bound_cap_for_pool
+                .floor()
+                .try_into()
+                .map_err(|e: anyhow::Error| ApiError::InternalError(e.to_string()))?
+        };
```

Comment:
I would also suggest doing this calculation precisely in `u128`.

Bₚ = floor( (κ * (T - Dₚ) - Cₚ) / (1 - κ) )

If you just multiply the top and bottom by 100,000 then the computation can't overflow a `u128`:

```
  (current_chain_parameters.capital_bound as u128 * (total_stake - delegated_stake_of_pool) as u128 - 100000 * (self.staked as u128))
  / (100000 - current_chain_parameters.capital_bound as u128)
```